contracts/Remittance.sol

```diff
@@ -1,38 +1,40 @@
 pragma solidity >=0.4.21 <0.6.0;
 
 import "./Pausable.sol";
-import "./Ownable.sol";
 
 contract Remittance is Pausable {
-  address public carol;
-  bytes32 private withdrawHash;
+  struct Escrow {
+    address carol;
+    uint256 amount;
+  }
+
+  mapping (bytes32 => Escrow) public escrows;
 
   event LogDeposited(address indexed sender, uint256 amount, bytes32 hash, address indexed carol);
-  event LogWithdrawn(address indexed sender, uint256 amount);
+  event LogWithdrawn(address indexed sender, bytes32 hash, uint256 amount);
```

Comment:
The hash is the guiding principle of the lifecycle. What addition would make sense?

Reply:
Shall we add Alice’s address here? Then we don’t need to look up Alice in LogDeposited event via the hash

Reply:
It is better to add the parameters where they make sense. If there is only a deposit event and it is missing Alice, how do you know what happened?